From 40618fdd452590cce7f02f0877785ee4031c4efa Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 12:25:59 -0400
Subject: [ethernet] tg3: Use pci_enable_msix_range() instead of pci_enable_msix()

Message-id: <1410179173-14304-6-git-send-email-ivecera@redhat.com>
Patchwork-id: 91670
O-Subject: [RHEL7.1 PATCH 05/19] tg3: Use pci_enable_msix_range() instead of pci_enable_msix()
Bugzilla: 1088637
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: #1088637

Upstream commit(s):
commit 6f1f411a2398bfdb55b4de9e60e830d5a85702d6
Author: Alexander Gordeev <agordeev@redhat.com>
Date:   Tue Feb 18 11:07:55 2014 +0100

    tg3: Use pci_enable_msix_range() instead of pci_enable_msix()

    As result of deprecation of MSI-X/MSI enablement functions
    pci_enable_msix() and pci_enable_msi_block() all drivers
    using these two interfaces need to be updated to use the
    new pci_enable_msi_range() and pci_enable_msix_range()
    interfaces.

    Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
    Cc: Nithin Nayak Sujir <nsujir@broadcom.com>
    Cc: Michael Chan <mchan@broadcom.com>
    Cc: netdev@vger.kernel.org
    Cc: linux-pci@vger.kernel.org
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/tg3.c b/drivers/net/ethernet/broadcom/tg3.c
index 9951c91..026cb18 100644
--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -11374,12 +11374,10 @@ static bool tg3_enable_msix(struct tg3 *tp)
   msix_ent[i].vector = 0;
  }
 
- rc = pci_enable_msix(tp->pdev, msix_ent, tp->irq_cnt);
+ rc = pci_enable_msix_range(tp->pdev, msix_ent, 1, tp->irq_cnt);
  if (rc < 0) {
   return false;
- } else if (rc != 0) {
-  if (pci_enable_msix(tp->pdev, msix_ent, rc))
-   return false;
+ } else if (rc < tp->irq_cnt) {
   netdev_notice(tp->dev, "Requested %d MSI-X vectors, received %d\n",
          tp->irq_cnt, rc);
   tp->irq_cnt = rc;
-- 
1.7.1